### ISSUE TITLE
ci: add Azure Static Web Apps workflow file Azure Static Web Apps CI/CD #1

### ISSUE THREAD

**Issue by jeemyng** [microsoft/Oryx#1506]
### Bug Report

- At what date and time did you most recently experience the problem?
> Just now

- Where did you experience the problem? E.g. Azure Web Apps, Azure Functions, Azure Container Registry, or offline use.
> Azure Web apps

- If your repo is publicly available please share its URL:
> https://github.com/jeemyng/hugo-static-app/commit/e6b8ae3d151c77a48450208915ab61d3c416b854

- What happened?
> I'm trying to publish a hugo site to azure static web apps. However, one of the deployed jobs for Azure Static Web Apps CI/CD was not successful.

- What did you expect or want to happen?
> Want the website to run and being able to open my deployed application. 

- How can we reproduce it?

- Do you have log output? Please include between the backticks:

```
Run Azure/static-web-apps-deploy@v1
/usr/bin/docker run --name cd9[8](https://github.com/jeemyng/hugo-static-app/runs/7740956038?check_suite_focus=true#step:4:9)fb6[9](https://github.com/jeemyng/hugo-static-app/runs/7740956038?check_suite_focus=true#step:4:10)12cd0fdf44a4a97114eea854e28b5_b18ac1 --label 4cd98f --workdir /github/workspace --rm -e INPUT_AZURE_STATIC_WEB_APPS_API_TOKEN -e INPUT_REPO_TOKEN -e INPUT_ACTION -e INPUT_APP_LOCATION -e INPUT_API_LOCATION -e INPUT_OUTPUT_LOCATION -e INPUT_API_BUILD_COMMAND -e INPUT_APP_ARTIFACT_LOCATION -e INPUT_APP_BUILD_COMMAND -e INPUT_ROUTES_LOCATION -e INPUT_SKIP_APP_BUILD -e INPUT_CONFIG_FILE_LOCATION -e INPUT_SKIP_API_BUILD -e INPUT_PRODUCTION_BRANCH -e INPUT_DEPLOYMENT_ENVIRONMENT -e INPUT_IS_STATIC_EXPORT -e HOME -e GITHUB_JOB -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_REPOSITORY_OWNER -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_RETENTION_DAYS -e GITHUB_RUN_ATTEMPT -e GITHUB_ACTOR -e GITHUB_TRIGGERING_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_SERVER_URL -e GITHUB_API_URL -e GITHUB_GRAPHQL_URL -e GITHUB_REF_NAME -e GITHUB_REF_PROTECTED -e GITHUB_REF_TYPE -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e GITHUB_ACTION_REPOSITORY -e GITHUB_ACTION_REF -e GITHUB_PATH -e GITHUB_ENV -e GITHUB_STEP_SUMMARY -e RUNNER_OS -e RUNNER_ARCH -e RUNNER_NAME -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/hugo-static-app/hugo-static-app":"/github/workspace" 4cd98f:b6912cd0fdf44a4a97114eea854e28b5
DeploymentId: cb2e64ca-6e34-490f-ac1e-7a589d7c33c3

App Directory Location: '/' was found.
No Api directory specified. Azure Functions will not be created.
Looking for event info
Starting to build app with Oryx
Azure Static Web Apps utilizes Oryx to build both static applications and Azure Functions. You can find more details on Oryx here: https://github.com/microsoft/Oryx
---Oryx build logs---


Operation performed by Microsoft Oryx, https://github.com/Microsoft/Oryx
You can report issues at https://github.com/Microsoft/Oryx/issues

Oryx Version: 0.2.20220401.1, Commit: afed4f696a08bfd4c8e4[10](https://github.com/jeemyng/hugo-static-app/runs/7740956038?check_suite_focus=true#step:4:11)9704a082bf9fb205[15](https://github.com/jeemyng/hugo-static-app/runs/7740956038?check_suite_focus=true#step:4:16), ReleaseTagName: 20220401.1

Build Operation ID: |60Aswmivqx0=.97c9be3f_
Repository Commit : e6b8ae3d151c77a48450208915ab61d3c4[16](https://github.com/jeemyng/hugo-static-app/runs/7740956038?check_suite_focus=true#step:4:17)b854

Detecting platforms...
Could not detect any platform in the source directory.
Error: Could not detect the language from repo.


---End of Oryx build logs---
Oryx was unable to determine the build steps. Continuing assuming the assets in this folder are already built. If this is an unexpected behavior please contact support.
Finished building app with Oryx
Failed to find a default file in the app artifacts folder (/). Valid default files: index.html,Index.html.
If your application contains purely static content, please verify that the variable 'app_location' in your workflow file points to the root of your application.
If your application requires build steps, please validate that a default file exists in the build output directory.


For further information, please visit the Azure Static Web Apps documentation at https://docs.microsoft.com/en-us/azure/static-web-apps/
If you believe this behavior is unexpected, please raise a GitHub issue at https://github.com/azure/static-web-apps/issues/
Exiting
```

**Comment by qianz2** [comment]
Hi @jeemyng ,

The error message shows that Oryx couldn't detect the hugo app from your repo. Please make sure that you're pointing to the correct app folder. And the Hugo toolset is run when one of following conditions met:
1. `config.yaml`, `config.toml`, `config.yml` or `config.json` configuration files in root of repo.
2. Recursively look up config directory: `config/**/*.yaml`, `config/**/*.toml`,
   `config/**/*.yml` or `config/**/*.json` configuration files in sub-directories.

**Comment by jeemyng** [comment]
Sorry, I do not understand your solution. Can you provide a step-by-step instruction for me? I am still learning and really appreciate your assistance

**Comment by qianz2** [comment]
Hi @jeemyng , the repo link you provided could not be found. Would you mind sharing the structure of your app repository? I'm trying to see if your repo met one of conditions above, and understand why Oryx couldn't detect hugo platform from your repo.

**Comment by jeemyng** [comment]
Here you go: https://github.com/jeemyng/hugo-static-app/blob/4c0596abe1272d9041ebd0db48e5705027e70d2c/.github/workflows/azure-static-web-apps-blue-bush-0dbd0bb00.yml

**Comment by cpdohert** [comment]
Hugo recently changed the default config file name from `config.toml` to `hugo.toml`; this appears to be breaking any build for Hugo > 0.110, and there's no way to override it in the `Azure/static-web-apps-deploy@v1` action.

Oryx is way, way too much an undocumented black box.

**Comment by malte610** [comment]
Yes. I encountered this problem as well. I have a hugo app created with 0.113.0 . Oryx couldn't detect a language. I included a config.toml file that contains my current hugo.toml and Oryx detected Hugo version 0.96.0 ☹️

**Comment by simonjj** [comment]
This should be fixed as of Jun 2023 with commit a4feb354923b8e05777839fffe57a03f2a1cdcbc. Will be closing this issue. If this is still relevant and impacting you please open a new issue and link back to this issue for reference.
